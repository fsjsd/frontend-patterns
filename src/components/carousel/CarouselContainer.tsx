import React, { Suspense } from 'react'
import { ContentWrapper } from '../../ux/ContentWrapper'
import Loading from '../../ux/Loading'
import Carousel, { CarouselSlideData } from './Carousel'

const urlRoot = process.env.REACT_APP_ROOT_URL

const CarouselContainer = () => {
  const data: CarouselSlideData[] = [
    {
      bannerText: 'Tour the Empire State Building! Buy tickets now.',
      imgUrl: `${urlRoot}/images/newyork.jpg`,
      imgAlt: 'New York City skyline',
    },
    {
      bannerText: 'Ride the Shinkansen!',
      imgUrl: `${urlRoot}/images/city.jpg`,
      imgAlt: 'Tokyo skyline',
    },
    {
      bannerText: 'See penguins!',
      imgUrl: `${urlRoot}/images/penguin.jpg`,
      imgAlt: 'Penguins',
    },
  ]

  return (
    <ContentWrapper
      title="Carousel"
      codeLink="/features/carousel"
      markDownPromise={import('./requirements.md')}
      noPadding={true}>
      <Suspense fallback={<Loading />}>
        <Carousel data={data} width={'1200'} />
      </Suspense>
    </ContentWrapper>
  )
}

export default CarouselContainer
