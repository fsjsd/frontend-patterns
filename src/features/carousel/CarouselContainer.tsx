import React from 'react'
import Carousel, { CarouselSlideData } from './Carousel'

const CarouselContainer = () => {

  const data: CarouselSlideData[] = [
    {
      bannerText: 'Tour the Empire State Building! Buy tickets now.',
      imgUrl: './images/newyork.jpg',
      imgAlt: 'New York City skyline',
    },
    {
      bannerText: 'Ride the Shinkansen!',
      imgUrl: './images/city.jpg',
      imgAlt: 'Tokyo skyline',
    },
    {
      bannerText: 'See penguins!',
      imgUrl: './images/penguin.jpg',
      imgAlt: 'Penguins',
    }
  ];

  return (
    <div>
      <Carousel data={data} width={"1200"} />
    </div>
  )
}

export default CarouselContainer