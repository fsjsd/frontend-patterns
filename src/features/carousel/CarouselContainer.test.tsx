import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import CarouselContainer from './CarouselContainer';

describe('CarouselContainer tests', () => {
  test('renders Carousel', () => {
    const { container } = render(<CarouselContainer />);
    expect(container).toMatchSnapshot();
  });
  test('forward advances slide', async () => {
    render(<CarouselContainer />);
    const slideContainer = screen.getByTestId("slidecontainer");
    const slide = screen.getByRole("tabpanel");
    expect(slide.offsetWidth).toBe(0);
    const firstTab = screen.getByRole('tab', { name: 'slide 1' });
    expect(firstTab.getAttribute("aria-selected")).toBe('true');
    expect(slideContainer.scrollLeft).toBe(0);
    const forwardButton = screen.getByRole('button', { name: 'forward' });
    fireEvent.click(forwardButton);
    expect(slideContainer.scrollLeft).toBe(-10);
    fireEvent.click(forwardButton);
    expect(slideContainer.scrollLeft).toBe(-20);
    fireEvent.click(forwardButton);
    expect(slideContainer.scrollLeft).toBe(0);
    //const secondTab = screen.getByRole('tab', { name: 'slide 2' });
    //expect(secondTab.getAttribute("aria-selected")).toBe('true');
  });

})
