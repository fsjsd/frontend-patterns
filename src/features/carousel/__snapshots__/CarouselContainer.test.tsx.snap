// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselContainer tests renders Carousel 1`] = `
<div>
  <div>
    <div
      class="sc-bdvvtL cPqwtb"
      width="1200"
    >
      <div
        class="sc-gsDKAQ gBTdBw"
        data-testid="slidecontainer"
      >
        <div
          aria-label="slide"
          class="sc-jRQBWg bXrUzc"
          data-slideindex="0"
          role="tabpanel"
        >
          <div
            class="sc-gKclnd iZYBQa"
          >
            Tour the Empire State Building! Buy tickets now.
          </div>
          <img
            alt="New York City skyline"
            height="600"
            src="/frontend-patterns/images/newyork.jpg"
            width="1200"
          />
        </div>
        <div
          aria-label="slide"
          class="sc-jRQBWg bXrUzc"
          data-slideindex="1"
          role="tabpanel"
        >
          <div
            class="sc-gKclnd iZYBQa"
          >
            Ride the Shinkansen!
          </div>
          <img
            alt="Tokyo skyline"
            height="600"
            src="/frontend-patterns/images/city.jpg"
            width="1200"
          />
        </div>
        <div
          aria-label="slide"
          class="sc-jRQBWg bXrUzc"
          data-slideindex="2"
          role="tabpanel"
        >
          <div
            class="sc-gKclnd iZYBQa"
          >
            See penguins!
          </div>
          <img
            alt="Penguins"
            height="600"
            src="/frontend-patterns/images/penguin.jpg"
            width="1200"
          />
        </div>
      </div>
      <div
        aria-label="back"
        class="sc-dkPtRN iynlET"
        role="button"
      >
        ←
      </div>
      <div
        aria-label="forward"
        class="sc-dkPtRN jpBcBY"
        role="button"
      >
        →
      </div>
      <div
        class="sc-hKwDye jSmFYF"
      >
        <div
          aria-label="slide 1"
          aria-selected="true"
          class="sc-eCImPb ePbVds"
          role="tab"
        />
        <div
          aria-label="slide 2"
          aria-selected="false"
          class="sc-eCImPb cJGTRq"
          role="tab"
        />
        <div
          aria-label="slide 3"
          aria-selected="false"
          class="sc-eCImPb cJGTRq"
          role="tab"
        />
      </div>
    </div>
  </div>
</div>
`;
